src/libraries/System.ObjectModel/src/System/Collections/ObjectModel/KeyedCollection.cs

```diff
@@ -280,5 +280,53 @@ private void RemoveKey(TKey key)
                 keyCount--;
             }
         }
+
+        internal sealed class KeyedCollectionDebugView
+        {
+            private readonly KeyedCollection<TKey, TItem> _collection;
+
+            public KeyedCollectionDebugView(KeyedCollection<TKey, TItem> collection)
+            {
+                ArgumentNullException.ThrowIfNull(collection);
+                _collection = collection;
+            }
+
+            [DebuggerBrowsable(DebuggerBrowsableState.RootHidden)]
+            public KeyedCollectionDebugViewItem[] Items
+            {
+                get
+                {
+                    var items = new KeyedCollectionDebugViewItem[_collection.Count];
+                    for (int i = 0; i < items.Length; i++)
+                    {
+                        TItem item = _collection[i];
+                        items[i] = new KeyedCollectionDebugViewItem(_collection.GetKeyForItem(item), item);
+                    }
+                    return items;
+                }
+            }
+        }
+
+        [DebuggerDisplay("{Value}", Name = "[{Key}]")]
+        internal readonly struct KeyedCollectionDebugViewItem
```

Comment:
private and nested inside of KeyedCollectionDebugView above?

Reply:
Tried making KeyedCollectionDebugViewItem nested inside KeyedCollectionDebugView  but got :

```
error CS0053: Inconsistent accessibility: property type 'KeyedCollection<TKey, TItem>.KeyedC
ollectionDebugView.KeyedCollectionDebugViewItem[]' is less accessible than property 'KeyedCollection<TKey, TItem>.KeyedCollectionDebugView.Items
```

Reply:
You'd fix that by making KeyedCollectionDebugViewItem public and nested. It wouldn't actually be visible outside of the assembly because it's contained inside something not public.